### ISSUE TITLE
[FEAT] Frequency based cross-validation and gap parameter

### ISSUE THREAD

**Issue by baggiponte** [neocortexdb/functime#89]
In the discord server, a while ago we mentioned the possibility of generating CV splits based on time intervals (e.g. the first day of the month or week). This might be useful for financial settings especially (e.g. retraining the model on mondays).

Plus, it might be useful to provide a `gap` parameter to allow for gaps between the train and test set (supported e.g. by [skforecast](https://skforecast.org/0.10.1/introduction-forecasting/introduction-forecasting#backtesting-including-gap)):

![image](https://github.com/neocortexdb/functime/assets/57922983/02605d29-5e49-4564-9a6b-359f23760eaa)

[plot source](https://fbruzzesi.github.io/timebasedcv/getting-started/#examples-of-cross-validation)

**Comment by elyase** [comment]
some additional references:

[sktime](https://www.sktime.net/en/stable/examples/forecasting/window_splitters.html): provides several splitting strategies: temporal_train_test_split (only sample space split), SlidingWindowSplitter, SlidingWindowSplitter, ExpandingWindowSplitter, CutoffSplitter (window can be a datetime)

[darts](https://unit8co.github.io/darts/generated_api/darts.utils.model_selection.html): only provides `train_test_split` but supports specifying the splits both in time and in sample space via an `axis` parameter

**Comment by FBruzzesi** [comment]
Hey @elyase, thanks for adding the suggestion.

As mentioned on discord, I was dealing with this kind of issue myself and came up with a rough implementation in [timebasedcv](https://github.com/FBruzzesi/timebasedcv) (some shameless promotion).

I wanted something that is scikit-learn compatible, hence supports (at least) numpy, pandas and polars.
I am not claiming it is the perfect solution by any mean, but I believe it brings a lot of the features that you are looking for.

Regarding integration with functime, I should double check how to make interaction with the `id/entity` column.
